includes/admin.form.inc

```diff
@@ -296,3 +1082,81 @@ function islandora_openseadragon_admin_submit($form, &$form_state) {
       break;
   }
 }
+
+/**
+ * Normalizes the settings so they can be used by Openseadragon.
+ *
+ * @param array $form
+ *   The Drupal form.
+ * @param array $form_state
+ *   The Drupal form state.
+ */
+function islandora_openseadragon_admin_submit_normalize(array $form, array &$form_state) {
+  _islandora_openseadragon_normalize_settings($form_state['values']['islandora_openseadragon_settings']);
+  _islandora_openseadragon_filter_settings($form_state['values']['islandora_openseadragon_settings']);
+}
+
+/**
+ * Casts the settings to appropriate types so they work in javascript.
+ *
+ * @param array $settings
+ *   The Openseadragon settings to be normalized.
+ *
+ * @return array
+ *   Normalized settings.
+ */
+function _islandora_openseadragon_normalize_settings(array &$settings) {
+  foreach ($settings as $key => $value) {
+    $settings[$key] = _islandora_openseadragon_normalize_setting($value);
+  }
+  return $settings;
+}
+
+/**
+ * Normalizes the given setting.
+ *
+ * @param mixed $value
+ *   The setting to be normalized.
+ *
+ * @return array|float|int|string
+ *   The normalized setting.
+ */
+function _islandora_openseadragon_normalize_setting($value) {
+  if (is_array($value)) {
+    return _islandora_openseadragon_normalize_settings($value);
+  }
+  elseif (filter_var($value, FILTER_VALIDATE_INT) !== FALSE) {
+    return (int) $value;
+  }
+  elseif (filter_var($value, FILTER_VALIDATE_FLOAT) !== FALSE) {
+    return (float) $value;
+  }
+  elseif (filter_var($value, FILTER_VALIDATE_URL) !== FALSE) {
+    return check_url($value);
```

Comment:
[`check_url()`](https://api.drupal.org/api/drupal/includes%21common.inc/function/check_url/7.x) has a fairly specific purpose... I'm not sure it fits the use here. Rare though they might be, should any of the characters which `check_plain()` escapes were to occur in the URL, it could well break wherever the values passing through here get used.